### PR TITLE
docs: Update test examples; add users guide

### PR DESCRIPTION
**Pull request by okurz**
* Use single ticks in general to prevent unexpected evaluation of special
  characters
* Use at least default timeout for assert_screen unless necessary
* Use `assert_script_run` instead of manual writing to serial port for
  checking status of command execution
* Use more explicit needle tag names
* Delete some redundant or empty statements

Related progress issue: https://progress.opensuse.org/issues/10380

### CODE REVIEW

**Comment by coolo** [comment]
do you have a link with the rendered docu?

File: docs/WritingTests.asciidoc [hunk 1]
```diff
@@ -90,21 +92,23 @@ sub test_flags {
 [source,perl]
 ----------------------------------------------------------------------------------------------------------
 sub run() {
-
     # change to root
     become_root;
 
     # output zypper repos to the serial
-    script_run "zypper lr -d > /dev/$serialdev";
+    script_run 'zypper lr -d > /dev/$serialdev';
```
Comment (aaannz) [inline docs/WritingTests.asciidoc]: This won't work, `$serialdev` is supposed to be replaced by perl.

Comment (coolo) [inline docs/WritingTests.asciidoc]: good catch

Comment (okurz) [inline docs/WritingTests.asciidoc]: fixed

**Comment by okurz** [comment]
see https://w3.nue.suse.com/~okurz/openQA_docs/ for rendered docu.

Would be nice if our travis job can also provide the rendered docu for reviewing automatically

@foursixnine

**Comment by coolo** [comment]
doing it in PR jobs is kind of impossible. Travis can only push to os-autoinst/openQA's gh-pages

**Comment by coolo** [comment]
and we would require a proper repository setup from everyone doing a PR

**Comment by okurz** [comment]
at least we can do what I did locally: Run asciidoc on all files in docs and 
provide that somehow

**Comment by coolo** [comment]
the somehow is the problem

**Comment by okurz** [comment]
2/5 tests fail. I assume some kind of parallel access problem in the tests? I will rebase and retry

**Comment by okurz** [comment]
rebased

**Comment by coolo** [comment]
redo the rebase - I removed the chrome UI tests now. Too flaky

**Comment by okurz** [comment]
rebased

**Comment by okurz** [comment]
@aaannz would you merge? mkittler would like to base on this and add some doc to UsersGuide.asciidoc

**Pull request merged**